corehq/apps/domain/test_decorators.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+from django.test.client import Client
+from corehq.util.test_utils import flag_enabled
+from corehq.apps.users.models import CouchUser
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.domain.decorators import _two_factor_required
+
+class TestDecorators(TestCase):
+
+    def setUp(self):
+        self.domain = create_domain("test_domain")
+        self.domain.two_factor_auth = False
+        self.request_2fa_enabled = self.create_request(request_url="/account/",
+                                                       username="test0@test.com",
+                                                       password="123",
+                                                       )
+        self.request_2fa_disabled = self.create_request(request_url="/account/",
+                                                       username="test_1@test.com",
+                                                       password="123",
+                                                       )
+
+    def create_request(self, request_url, username, password):
+        # Initialize request
+        request = Client().get(request_url).wsgi_request
+        # Create user
+        request.user = get_user_model().objects.create_user(username=username, email=username,
+                                                            password=password)
+        # Create couch user
+        request.couch_user = CouchUser()
+        # Login
+        assert Client().login(username=username, password=password)
+        return request
+
+    @classmethod
+    def enable_two_factor_for_user(cls, request):
+        request.user.otp_device = "test_device"
```

Comment:
how is this enabling 2fa for the user? Not sure I understand what this is doing...